specifications/api/swagger.yaml

```diff
@@ -321,6 +326,30 @@ x-poolMetadataSource: &poolMetadataSource
       format: uri
       example: https://smash.cardano-mainnet.iohk.io/
 
+x-walletCredentialPubkey: &walletCredentialPubkey
+  description: An extended public key (public key + chain code) for credential
+  type: string
+  format: hex
+  minLength: 128
+  maxLength: 128
+  example: 1423856bc91c49e928f6f30f4e8d665d53eb4ab6028bd0ac971809d514c92db11423856bc91c49e928f6f30f4e8d665d53eb4ab6028bd0ac971809d514c92db1
+
+x-walletScript: &walletScript
+  description: A script as text with predicates and verification keys. Here probably we should add a full-fledged script object
+  type: string
```

Comment:
How is that a `string` :thinking: ? Isn't the goal here to re-use the JSON object instances from cardano-addresses?

Reply:
it is going to be json object. just doing this in steps

Reply:
my attempt -> https://github.com/input-output-hk/cardano-wallet/pull/2253/commits/ccd4eb486e7739d6a71497891baf70b0f78eff86
But when
```
$ openapi-spec-validator --schema 3.0.0 specifications/api/swagger.yaml 
found undefined alias 'walletCredentialScript'
  in "<file>", line 357, column 14
```

Reply:
Look at transaction metadatum. I recall we did something similar.

Reply:
thanks, now it seems to be ok -> https://github.com/input-output-hk/cardano-wallet/pull/2253/commits/2f9180366bdb4319b19c3ad90bb06e5f52fc350e